eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1

```diff
@@ -215,9 +216,56 @@ function DeployStressPackage(
     if ($LASTEXITCODE) {exit $LASTEXITCODE}
 
     $dockerBuildConfigs = @()
-    
+
     $genValFile = Join-Path $pkg.Directory "generatedValues.yaml"
     $genVal = Get-Content $genValFile -Raw | ConvertFrom-Yaml -Ordered
+    $releaseName = $pkg.ReleaseName
+    if ($RerunFailedJobs) {
+        $pods = kubectl get pods -n $pkg.namespace -o json | ConvertFrom-Json
+
+        # Get all jobs within this helm release
+        $helmResources = helm status -n $pkg.Namespace $pkg.ReleaseName --show-resources
+        $discoveredJob = $False
+        $jobs = @()
+        foreach ($helmResource in $helmResources) {
+            if ($discoveredJob -and $helmResource -match "==>") {break}
+            if ($discoveredJob) {
+                $jobs += ($helmResource -split '\s+')[0] | where{($_ -ne "NAME") -and ($_)}
```

Comment:
```suggestion
                $jobs += ($helmResource -split '\s+')[0] | Where-Object {($_ -ne "NAME") -and ($_)}
```

Reply:
what's the advantage of doing so?

Reply:
"where" seems to only work on arrays & will return empty array instead of null when no matches were found

Reply:
We favor using full cmdlet names to reduce confusion (see [explanation](https://github.com/Azure/azure-sdk-tools/blob/main/doc/development/powershell.md#:~:text=Use%20full%20CmdLet%20names%20for%20better%20scripts)). For example, `where` as it is used here is an alias for `Where-Object` and is specifically different from `.Where()` which I think you are describing?